### PR TITLE
kv_filter expansion

### LINKED ISSUE #491: ENH: filter and print key functions not so tightly coupled.
Basically, I kind of want the powerful kv_filtering available but then do other interesting things with the filtered data than just print ids or a key value.  Can we integrate the kv_filtering into the main flow?

### PR DESCRIPTION
**Pull request by mwinitch**
Closes #491

### CODE REVIEW

**Comment by mwinitch** [comment]
I have modified `l_members` to show what I think could be the functionality you want. Basically I have made it so if the filter flags are given the new valid collection becomes the database the lister goes through, otherwise, the lister goes through the default collection. If this looks good to you I can modify the other listers in a similar way.

File: regolith/helpers/l_membershelper.py [hunk 1]
```diff
@@ -68,16 +69,21 @@ def construct_global_ctx(self):
     def sout(self):
         rc = self.rc
         if rc.filter:
-            results = search_collection(self.gtx["people"], rc.filter, rc.keys)
-            print(results, end="")
-            return
+            database = key_value_pair_filter(self.gtx["people"], rc.filter)
```
Comment (sbillinge) [inline regolith/helpers/l_membershelper.py:72]: This is just right!   super!

I think this function returns a collection, not a database.  Let's make sure we use the jargon correctly to both avoid confusing people and also training them in the jargon.

File: regolith/helpers/l_membershelper.py [hunk 2]
```diff
@@ -68,16 +69,21 @@ def construct_global_ctx(self):
...
         bad_stati = ["finished", "cancelled", "paused", "back_burner"]
         people = []
-        for person in self.gtx["people"]:
+        for person in database:
             if rc.current and not person.get('active'):
                 continue
             people.append(person)
-        #people.sort()
+
+        if rc.filter and not rc.verbose:
```
Comment (sbillinge) [inline regolith/helpers/l_membershelper.py:82]: yes, I think this will work too.

**Comment by mwinitch** [comment]
@sbillinge What are your thoughts on the updates?

**Comment by sbillinge** [comment]
These look good.  Is it ready for merge (don't forget to remove WIP and ask for the merge)

**Comment by mwinitch** [comment]
@sbillinge This is ready for review and merge

**Comment by sbillinge** [comment]
thanks @mwinitch !

**Pull request merged**